Add E2E tests for the Accordion block

louwie17 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Adds E2E tests for the accordion block and updates the textdomain of the accordion blocks.

Closes #54666 

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build this branch and set up your docker environment: `pnpm --filter=@woocommerce/block-library env:start`
2. Run the new E2E tests: `pnpm --filter=@woocommerce/block-library test:e2e accordion/accordion.block_theme.spec.ts` and make sure they pass.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

louwie17 committed:
Add e2e tests for Accordion block functionality

- Implement tests to verify Accordion block behavior with feature flags.
- Ensure Accordion block cannot be inserted when the feature flag is disabled.
- Validate that the Accordion block is visible on the frontend when the feature flag is enabled.
- Test adding titles and content to Accordion panels.
- Check panel visibility toggling and default open settings.
- Verify auto-close functionality when multiple panels are used.

Includes new test and page object files for Accordion block interactions.

louwie17 committed:
Update textdomain for Accordion block JSON files to 'woocommerce'

louwie17 committed:
Add changelog

gigitux commented on plugins/woocommerce-blocks/tests/e2e/tests/accordion/accordion.block_theme.spec.ts:
```diff
@@ -0,0 +1,241 @@
+		requestUtils,
+	} ) => {
+		await requestUtils.setFeatureFlag( 'experimental-blocks', false );
+		await admin.createNewPost();
```
I noticed that this code is duplicated in all the tests. What are your thoughts on moving these steps in `beforeEach` hook?

louwie17 replied:
Good suggestion, done as part of: https://github.com/woocommerce/woocommerce/pull/54739/commits/f4b0e82c1267681d91a4513de19574edb79517df

gigitux commented on plugins/woocommerce-blocks/tests/e2e/tests/accordion/accordion.block_theme.spec.ts:
```diff
@@ -0,0 +1,241 @@
+				title: 'Accordion title 2',
+				content: 'Test paragraph content for second panel',
+			},
+		] );
```
Given that we will rely on `beforeEach` hook for adding the block, it could be necessary to change the `insertAccordionGroup` logic: we could pass the `blockLocator` instance as argument.

louwie17 replied:
I end up leaving the `insertAccordionGroup` within each test, as they weren't all strictly unique.

gigitux reviewed (approved):
LGTM! I suggested a minor change to improve the readability of the test, but feel free to skip!

louwie17 committed:
Move some logic to a beforeEach

louwie17 commented:
> LGTM! I suggested a minor change to improve the readability of the test, but feel free to skip!

Thanks for the review @gigitux, I addressed some of your suggestions. I added the `beforeEach`, but didn't move the insert block logic in there. Would you be able to give this another look?
